### PR TITLE
Fix for messages behavior depends on 'fileformat' of current buffer

### PR DESCRIPTION
**Pull request by cero1988**

### CODE REVIEW

**Comment by brammool** [comment]
Please sync to head to avoid the tests failing.

File: src/charset.c [hunk 1]
```diff
@@ -527,26 +527,31 @@ transchar_buf(buf_T *buf, int c)
  */
     char_u *
 transchar_byte(int c)
+{
+    return transchar_byte_buf(curbuf, c);
+}
+
+    char_u *
+transchar_byte_buf(buf_T *buf,int c)
```
Comment (k-takata) [inline src/charset.c]:
```suggestion
transchar_byte_buf(buf_T *buf, int c)
```

File: src/charset.c [hunk 2]
```diff
@@ -527,26 +527,31 @@ transchar_buf(buf_T *buf, int c)
...
 {
     if (enc_utf8 && c >= 0x80)
     {
-	transchar_nonprint(curbuf, transchar_charbuf, c);
-	return transchar_charbuf;
+        transchar_nonprint(buf, transchar_charbuf, c);
+        return transchar_charbuf;
```
Comment (k-takata) [inline src/charset.c]:
```suggestion
	transchar_nonprint(buf, transchar_charbuf, c);
	return transchar_charbuf;
```
Tab characters should be used here.

File: src/charset.c [hunk 3]
```diff
@@ -527,26 +527,31 @@ transchar_buf(buf_T *buf, int c)
...
     }
-    return transchar(c);
+    return transchar_buf(buf, c);
 }
-
 /*
  * Convert non-printable character to two or more printable characters in
- * "charbuf[]".  "charbuf" needs to be able to hold five bytes.
+ * "buf[]".  "charbuf" needs to be able to hold five bytes.
```
Comment (k-takata) [inline src/charset.c]: I don't think this change is needed.

The following might be better?
```
 * Convert non-printable character "c" to two or more printable characters
 * in "charbuf[]".  "charbuf" needs to be able to hold five bytes.
```

File: src/charset.c [hunk 4]
```diff
@@ -527,26 +527,31 @@ transchar_buf(buf_T *buf, int c)
...
  * Does NOT work for multi-byte characters, c must be <= 255.
  */
     void
 transchar_nonprint(buf_T *buf, char_u *charbuf, int c)
 {
     if (c == NL)
 	c = NUL;		// we use newline in place of a NUL
-    else if (c == CAR && get_fileformat(buf) == EOL_MAC)
+    else if (buf != NULL && c == CAR && get_fileformat(buf) == EOL_MAC)
 	c = NL;			// we use CR in place of  NL in this case
```
Comment (k-takata) [inline src/charset.c:555]:
```diff
-	c = NL;			// we use CR in place of  NL in this case
+	c = NL;			// we use CR in place of NL in this case
```
This is not your change, but there is an extra space in the comment.

Comment (cero1988) [reply src/charset.c:555]: thank you for the suggestions, I have commited them all

File: src/charset.c [hunk 5]
```diff
@@ -527,15 +527,20 @@ transchar_buf(buf_T *buf, int c)
...
     if (enc_utf8 && c >= 0x80)
     {
-	transchar_nonprint(curbuf, transchar_charbuf, c);
+	transchar_nonprint(buf, transchar_charbuf, c);
 	return transchar_charbuf;
     }
-    return transchar(c);
+    return transchar_buf(buf, c);
 }
-
```
Comment (k-takata) [inline src/charset.c:538]: This empty line should be kept.

**Pull request closed without merging**